modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,427 @@
+
+#define STOP_MODE "stop"
+#define DRAIN_MODE "drain"
+#define PUMP_MODE "pump"
+
+////
+//	The item in its functional state
+/obj/structure/bed/borg_action_pacifier
+	name = "Deployed B.A.P. unit"
+	desc = "An inactivated device to constrain silicons with."
+	icon = 'modular_skyrat/modules/deployables/icons/deployable.dmi'
+	icon_state = "up"
+	anchored = FALSE
+	resistance_flags = FIRE_PROOF | FREEZE_PROOF
+	flags_1 = NODECONSTRUCT_1
+
+	bolts = FALSE
+	// The cyborg currently buckled to the BAP
+	var/mob/living/silicon/robot/buckled_cyborg
+	// If the BAP is deployed or not
+	var/deployed = TRUE
+	// Wether or not the machine is locking the cyborg
+	var/locked = FALSE
+	// To distinguish if the machine is pumping or draining
+	var/enabled_function = NONE
+	// Amount of power drained from the cyborg, which we are now storing
+	var/power_storage = 0
+
+/obj/structure/bed/borg_action_pacifier/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CLICK_ALT, .proc/check_alt_clicked_radial)
+
+/obj/structure/bed/borg_action_pacifier/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_CLICK_ALT)
+
+/obj/structure/bed/borg_action_pacifier/examine(mob/user)
+	. = ..()
+	if(locked)
+		. += span_notice("It's locked.")
+	if(enabled_function != NONE)
+		. += span_notice("It's [enabled_function]ing power...")
+	. += span_notice("It's currently holding [power_storage] units worth of charge.")
+	if(power_storage == 40000)	// BS cell capacity, a potato battery would still outplay this
+		. += span_warning("It cannot store any more power.")
+
+////
+//	The grenade
+/obj/item/grenade/borg_action_pacifier_grenade
+	name = "B.A.P. grenade"
+	desc = "An inactivated device to constrain silicons with."
+	icon = 'modular_skyrat/modules/deployables/icons/deployable.dmi'
+	icon_state = "folded"
+	inhand_icon_state = "folded"
+	worn_icon_state = "folded"
+	w_class = WEIGHT_CLASS_NORMAL
+
+	det_time = 3 SECONDS
+	// Amount of power drained from the cyborg, from when we were still deployed
+	var/power_storage = 0
+
+/obj/item/grenade/borg_action_pacifier_grenade/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CLICK_ALT, .proc/check_alt_clicked_grenade)
+
+/obj/item/grenade/borg_action_pacifier_grenade/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_CLICK_ALT)
+
+/obj/item/grenade/borg_action_pacifier_grenade/detonate(mob/living/lanced_by)
+	. = ..()
+	if(!.)
+		return
+
+	var/obj/structure/bed/borg_action_pacifier/undeployed/deploying/deploying = new /obj/structure/bed/borg_action_pacifier/undeployed/deploying(get_turf(src.loc))
+	deploying.power_storage = power_storage
+	qdel(src)
+
+/obj/item/grenade/borg_action_pacifier_grenade/examine(mob/user)
+	. = ..()
+	. += span_notice("It's currently holding [power_storage] units worth of charge.")
+	if(power_storage == 40000)
+		. += span_warning("It cannot store any more power.")
+
+////
+//	The state after popping out of the grenade
+/obj/structure/bed/borg_action_pacifier/undeployed/deploying
```

Comment:
This is generally a pretty awful way of doing this. Have it create the regular BAP but do a flick() to animate the icon state (or animate())